feat: Enhanced admin dashboard

DedsecKnight opened the pull request:
Changes made in this PR: 
- Added a new dashboard to allow toggling membership status
- Added a new dashboard to allow adding/deleting directors. 
- Added a new dashboard to display event statistics.

DedsecKnight committed:
feat: added new graphql resolvers for query event reservations info

DedsecKnight committed:
feat: added event participation dashboard

DedsecKnight committed:
fix: updated type definition to prevent build error

DedsecKnight committed:
refactor: added back tailwindcss config removed by shadcn

DedsecKnight committed:
feat: added director management dashboard

DedsecKnight committed:
feat: added member management dashboard

DedsecKnight committed:
fix: fixed logic bug in adding officer into a division

DedsecKnight committed:
feat: added fuzzy search into event statistics management

DedsecKnight committed:
feat: added fuzzy search into director management view

DedsecKnight committed:
feat: redo member management dashboard and added fuzzy search

DedsecKnight committed:
refactor: added guard for data fetching

DedsecKnight committed:
chore: removed redundant imports

DedsecKnight committed:
perf: added dataloader to improve querying complexity

DedsecKnight committed:
feat: restricted access to certain resolvers to dev director or executive only

DedsecKnight committed:
refactor: revert css config changes

DedsecKnight committed:
fix: fixed build error

DedsecKnight committed:
feat: added button to choose a random event attendee

arihants04 reviewed (approved):
looks good everything works!

DedsecKnight committed:
refactor: enforced restriction on director management pages

Restricted access to director management pages to dev director &
executives only

DedsecKnight committed:
Merge branch 'next' into feat/admin_dashboard
